### PR TITLE
Add smoke test to PR workflow

### PR DESCRIPTION
**Pull request by sophia-guo**
Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

**Comment by gdams** [comment]
I'd suggest making the smoke test part of the build rather than redefining the matrix again and creating a separate job?

File: .project [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>openjdk-build</name>
```
Comment (gdams) [inline .project:3]: is this file needed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -188,3 +279,46 @@ jobs:
        with:
          name: ${{matrix.version}}-${{matrix.os}}-${{matrix.vm}}
          path: workspace/target/*
+  
+  windows_smokeTest:
+    name: macos_smokeTest
+    runs-on: macos-latest
```
Comment (gdams) [inline .github/workflows/build.yml:285]: assume this is meant to run on `windows-latest`?

File: .project (hunk 1 above)
Comment (sophia-guo) [reply .project:3]: no, I will remove it.

**Comment by sophia-guo** [comment]
> I'd suggest making the smoke test part of the build rather than redefining the matrix again and creating a separate job?

Yes, we can do like that. The benefit of separating build and test job is to make it clear if test job fails while build job passes. However if we'd like smoke tests block the build jobs if there is a failure making tests part of the build would make more sense.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -192,3 +285,47 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Restore build artifacts
+      uses: actions/download-artifact@v2
+      with:
+        name: ${{matrix.version}}-${{matrix.os}}-${{matrix.vm}}
+        path: ~/${{matrix.version}}-${{matrix.os}}-${{matrix.vm}}
+    - name: Unpack jdk
+      run: |
+        tar -xf "${HOME}/${{matrix.version}}-${{matrix.os}}-${{matrix.vm}}/OpenJDK.tar.gz" -C "${HOME}/${{matrix.version}}-${{matrix.os}}-${{matrix.vm}}"
```
Comment (gdams) [inline .github/workflows/build.yml:313]: this is going to need to be an unzip command

**Comment by gdams** [comment]
> Yes, we can do like that. The benefit of separating build and test job is to make it clear if test job fails while build job passes. However if we'd like smoke tests block the build jobs if there is a failure making tests part of the build would make more sense.

Yeah i'd really prefer that it was part of the build, it will make it super clear when we make breaking changes to the build scripts if you're happy to merge the smoke jobs into the build ones?

File: .github/workflows/build.yml (hunk 3 above)
Comment (gdams) [reply .github/workflows/build.yml:313]: You likely need a rebase to pick up https://github.com/adoptium/temurin-build/commit/631f4e3d41b2c33fec83fad88a9264c8130b73d0

**Comment by sophia-guo** [comment]
re https://github.com/adoptium/temurin-build/pull/2660#issuecomment-863036615

Sure, will update to combine the test as a part of the build.

**Comment by gdams** [comment]
@sophia-guo this should fix the centos issues (looking for `apt-get` https://github.com/adoptium/run-aqa/pull/91

**Review by gdams — APPROVED** [review]

**Comment by gdams** [comment]
Linter failure can be ignored - there is no way to avoid the duplicate code

**Review by smlambert — APPROVED** [review]
LGTM
